src/main/java/com/sumi/smartemployeeservice/services/EmployeeService.java

```diff
@@ -0,0 +1,60 @@
+package com.sumi.smartemployeeservice.services;
+
+import com.sumi.smartemployeeservice.modal.Employee;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+@Service
+public class EmployeeService {
+
+    private List<Employee> employees = new ArrayList<>(Arrays.asList(
+            new Employee(1, "sumi", "katu"),
+
+            new Employee(2, "sumi", "katu")
+
+    ));
+
+
+    public List<Employee> getEmployees() {
+
+        return employees;
+    }
+
+    public Employee getEmployee(int id) {
+        Employee emp = new Employee();
+        for (Employee employee : employees
+        ) {
+            if (employee.getId() == id) {
+                emp = employee;
+            }
+        }
+        return emp;
+    }
+
+    public void createEmployee(Employee employee) {
+        employees.add(employee);
+    }
+
+    public void editEmployee(Employee employee, int id) {
+        Employee temp = new Employee();
+        for (int i = 0; i < employees.size(); i++) {
+
+            if (employees.get(i).getId() == id) {
+                employees.set(i, employee);
+            }
+        }
```

Comment:
you can see line 26 - 33 and 42 to 47, we can see code duplication. this a very good reason to create a private method.

Reply:
I can understand there is a code duplication. but can't understand how it can be resolve using private method

Reply:
write a private method (for the now) called getEmployeeById(). actually notice you have a similar public method. call that method when you want to find by id.

for ex. 

> public void editEmployee(Employee employee, int id) {
```suggestion
        Employee temp = new Employee();
        for (int i = 0; i < employees.size(); i++) {

            if (employees.get(i).getId() == id) {
                employees.set(i, employee);
            }
        }
```
>         Employee temp = getEmployee(id)
>         employees.set(i, employee);
> }